azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunnerManager.java

```diff
@@ -341,18 +369,30 @@ private FlowRunner createFlowRunner(final int execId) throws ExecutorManagerExce
           + execId);
     }
 
-    // Record the time between submission, and when the flow preparation/execution starts.
-    // Note that since submit time is recorded on the web server, while flow preparation is on
-    // the executor, there could be some inaccuracies due to clock skew.
-    commonMetrics.addQueueWait(System.currentTimeMillis() -
-        flow.getExecutableFlow().getSubmitTime());
-
-    final Timer.Context flowPrepTimerContext = commonMetrics.getFlowSetupTimerContext();
-    try {
-      // Sets up the project files and execution directory.
-      this.flowPreparer.setup(flow);
-    } finally {
-      flowPrepTimerContext.stop();
+    // Sets up the project files and execution directory.
+    if (this.active) {
```

Comment:
It would be possible to skip the outer "if (this.active)" check, since this is checked again after the increment, and would save the additional else clause, so would be more concise.

Reply:
The if check also needs to check if use executor has been set for the flow, and still execute if true.

Reply:
updated